features/lwipstack/LWIPInterface.cpp

```diff
@@ -753,6 +743,58 @@ nsapi_error_t LWIP::Interface::bringup(bool dhcp, const char *ip, const char *ne
     return NSAPI_ERROR_OK;
 }
 
+nsapi_error_t LWIP::Interface::set_ip_address(const char *ip,
+                                              const char *netmask,
+                                              const char *gw,
+                                              uint8_t ipv6_flag)
+{
+    if (!ip) {
+        return NSAPI_ERROR_PARAMETER;
+    }
+
+    int conv_ip = 1;
+
+#if LWIP_IPV4
```

Comment:
If both LWIP_IPV4 & 6 are enabled, can you still set IPv6 address? inet_aton returns NULL in case of wrong format? Maybe a comment would be needed here to make it clear?

Reply:
Yes you can. In one of our projects we set up static IPv4 address using `set_ip_address` API and then give link-local IPv6 address in `bringup`. And `bringup` internally uses the same `set_ip_address` to set IPv6 address.